### PR TITLE
[VPAT:Developer] Add link to bottom of pages in development to automatically validate html

### PR DESCRIPTION
**Pull request by elihschiff**
### Please check if the PR fulfills these requirements:

* [X] The PR title and message follows our [guidelines](https://submitty.org/developer/how_to_contribute)

### What is the current behavior?

Currently it is a huge pain to validate the html of submitty pages. It usually involves downloading the rendered html of a page and then uploading it to https://validator.w3.org/#validate_by_upload

### What is the new behavior?

This PR adds a button to the bottom of all pages (only in development mode) which allows you to in one click validate the html of a page
image

I will also be adding some documentation for this change and will link that pr here as well once it is done: https://github.com/Submitty/submitty.github.io/pull/231

### CODE REVIEW

**Comment by bmcutler** [comment]
@elihschiff 
I've installed the PR, but I don't see this link in my footer.
I wonder if it's conflicting with this other recently merged PR:
https://github.com/Submitty/Submitty/pull/5119
which edits the footer to add sysadmin contact info?

**Comment by elihschiff** [comment]
@bmcutler I just looked at that pr and I dont think it should conflict. I just tested this branch and I still see the button. Do you see the `Show Page Details` button at the bottom of your page? Is there a chance you are not in debug mode?
image

File: site/public/js/server.js [hunk 1]
```diff
@@ -806,6 +806,52 @@ function togglePageDetails() {
     }
 }
 
+/**
+ * Opens a new tab on https://validator.w3.org with the contents of the current html page
+ */
+function validateHtml() {
+  //Code copied from https://validator.w3.org/nu/about.html under "Check serialized DOM of current page" secton
+  function c(a, b) {
+    var c = document.createElement("textarea");
```
Comment (MasterOdin) [inline site/public/js/server.js:815]: Use `const` for variables that do not change

File: site/public/js/server.js [hunk 2]
```diff
@@ -806,6 +806,52 @@ function togglePageDetails() {
...
+    c.name = a;
+    c.value = b;
+    d.appendChild(c)
+  }
+  var e = function(a) {
```
Comment (MasterOdin) [inline site/public/js/server.js:820]:
```suggestion
  const e = (a) => {
```

File: site/public/js/server.js [hunk 3]
```diff
@@ -806,6 +806,52 @@ function togglePageDetails() {
...
+            b += "<\!--" + a.nodeValue + "--\>";
+            break;
+          case Node.DOCUMENT_TYPE_NODE:
+            b += "<!DOCTYPE " + a.name + ">\n"
+        }
+        a = a.nextSibling
+      }
+      return b
+    }(document),
+    d = document.createElement("form");
```
Comment (MasterOdin) [inline site/public/js/server.js:842]:
```suggestion
  }(document);
  const d = document.createElement("form");
```

**Comment by bmcutler** [comment]
Yup, debugging was disabled.
That's supposed to be enabled by default for vagrant, right?
Perhaps my testing of the sysadmin footer PR disabled debug mode?

**Pull request merged**